clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/ConnectionState.java

```diff
@@ -247,9 +249,13 @@ public void destroy(String name) throws CachePersistenceException {
           throw new CachePersistenceException("Could not connect to the cluster tier manager '" + entityIdentifier
                   + "'; retrieve operation timed out", e);
         } catch (DestroyInProgressException e) {
-          silentDestroy();
-          // Nothing left to do
-          break;
+          try {
```

Comment:
```
          try {
            silentDestroy();
            // Nothing left to do
            break;
          } catch (ConnectionClosedException | ConnectionShutdownException e1) {
            reconnect();
          }
```

This block of code is repeating at multiple places. See if you can extract it out to a method and reuse.